angular/src/app/users/reset-password/reset-password.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit, Optional, Injector, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+import { AppComponentBase } from '@shared/app-component-base';
+import { finalize } from 'rxjs/operators';
+import {
+  UserServiceProxy,
+  ResetPasswordDto
+} from '@shared/service-proxies/service-proxies';
+
+@Component({
+  selector: 'app-reset-password',
+  templateUrl: './reset-password.component.html'
+})
+export class ResetPasswordDialogComponent extends AppComponentBase
+  implements OnInit {
+  public isLoading = false;
+  public resetPasswordDto: ResetPasswordDto;
+
+  constructor(
+    injector: Injector,
+    private _userService: UserServiceProxy,
+    private _dialogRef: MatDialogRef<ResetPasswordDialogComponent>,
+    @Optional() @Inject(MAT_DIALOG_DATA) private _userId: number
+  ) {
+    super(injector);
+  }
+
+  ngOnInit() {
+    this.isLoading = true;
+    this.resetPasswordDto = new ResetPasswordDto();
+    this.resetPasswordDto.userId = this._userId;
+    this.resetPasswordDto.newPassword = Math.random()
+      .toString(36)
+      .substr(2, 10);
```

Comment:
Not the prettiest random password, but the goal was simply to ensure each user has a different password when they get reset so admins don't go using the same password for everyone.  Hopefully they give instructions for each user to reset their own password when done, or better the system could enforce that at some point.